filters/profilter.go

```diff
@@ -21,29 +21,12 @@ type LanternProFilter struct {
 	proTokens *set.Set
 }
 
-type optSetter func(f *LanternProFilter) error
-
-func Logger(l utils.Logger) optSetter {
-	return func(f *LanternProFilter) error {
-		f.log = l
-		return nil
-	}
-}
-
-func New(next http.Handler, setters ...optSetter) (*LanternProFilter, error) {
-	f := &LanternProFilter{
-		log:       utils.NullLogger,
+func NewProFilter(next http.Handler, log utils.Logger) *LanternProFilter {
```

Comment:
Removed `Logger` and `optSetter` to make code simpler. Same for below two filters.